tests/test_urlgrabber.py

```diff
@@ -158,3 +159,26 @@ def test_http_not_found(self):
             self._runner(self.httpbin.url + "/status/404", 404)
         with pytest.raises(urllib.error.HTTPError):
             self._runner(self.httpbin.url + "/no/such/file", 404)
+
+
+class TestFilenameFromDispositionHeader:
+    @pytest.mark.parametrize(
+        "header, result",
+        [
+            (
+                "filename=Zombie.Land.Saga.Revenge.S02E12.480p.x264-mSD.nzb; filename*=UTF-8''Zombie.Land.Saga.Revenge.S02E12.480p.x264-mSD-utf.nzb",
+                "Zombie.Land.Saga.Revenge.S02E12.480p.x264-mSD-utf.nzb"
+            ),
+            (
+                "filename=Zombie.Land.Saga.Revenge.S02E12.480p.x264-mSD.nzb;",
+                "Zombie.Land.Saga.Revenge.S02E12.480p.x264-mSD.nzb"
+            ),
+            (
+                "filename*=UTF-8''Zombie.Land.Saga.Revenge.S02E12.480p.x264-mSD.nzb",
+                "Zombie.Land.Saga.Revenge.S02E12.480p.x264-mSD.nzb"
+            ),
+        ]
+    )
```

Comment:
might want to add tests for corner cases too, e.g. input that causes the filename to come up empty and verify the function return correctly for those (i.e., `None` rather than an empty string).

Reply:
Other corner cases would be filenames that try to include path elements (/what/ever/file.name, ../../dir/escape.txt).

Reply:
I agree that we could check for some corner cases. But we should asume at some point that where parsing RFC compliant strings imo

Reply:
Never mind. I"ve added the "/what/ever/file.name" test. And it passes. It extracts the basename.